catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -22,6 +23,41 @@
     raise
 
 
+def _create_property(hamamatsu_property_name, read_only=False, stopped_acquisition=True):
+    def getter(self):
+        with self.mutex:
+            if hamamatsu_property_name == 'EXPOSURETIME':
+                return self.cam.prop_getvalue(getattr(dcam.DCAM_IDPROP, hamamatsu_property_name)) * 1e6
+            elif hamamatsu_property_name in ['IMAGE_WIDTH', 'IMAGE_HEIGHT', 'SUBARRAYHSIZE', 'SUBARRAYVSIZE', 'SUBARRAYVPOS', 'SUBARRAYHPOS']:
+                return int(self.cam.prop_getvalue(getattr(dcam.DCAM_IDPROP, hamamatsu_property_name)))
+            else:
+                return self.cam.prop_getvalue(getattr(dcam.DCAM_IDPROP, hamamatsu_property_name))
+
+    if read_only:
+        setter = None
+    else:
+        def setter(self, value):
+            was_running = self.is_acquiring.get()[0] > 0
+
+            if was_running and stopped_acquisition:
+                self.end_acquisition()
+
+                while self.is_acquiring.get()[0]:
+                    time.sleep(0.001)
+
+            with self.mutex:
+                if hamamatsu_property_name == 'EXPOSURETIME':
+                    self.cam.prop_setvalue(getattr(dcam.DCAM_IDPROP, hamamatsu_property_name), value / 1e6)
+                elif hamamatsu_property_name in ['IMAGE_WIDTH', 'IMAGE_HEIGHT', 'SUBARRAYHSIZE', 'SUBARRAYVSIZE', 'SUBARRAYVPOS', 'SUBARRAYHPOS']:
+                    int(self.cam.prop_setvalue(getattr(dcam.DCAM_IDPROP, hamamatsu_property_name), value))
```

Comment:
The PR is good to go in my opinion, but this `int()` makes no sense here. If the code works as-is then just delete it, since you're casting a function to an int. Or you're casting something (whatever is returned by `self.cam.prop_setvalue()`) to an int that you're not reusing anyway.

If you don't need to cast `value` to int before sending it to the camera (which is not being done right now), it means you could delete the entire `elif` clause here - if you have tested on hardware that all the setters actually work.

Reply:
ok, let me make this correction then retest on hardware then merge